script/build/libressl.asc

```diff
@@ -0,0 +1,182 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```

Comment:
why is this needed?

Reply:
To verify the signature of the libressl tarball we download.

Reply:
It may be easier (and equivalent from a security) to have a hash of the specific version you are downloading. I suspect getting that gpg to work properly on windows may not be that easy.

Reply:
> I suspect getting that gpg to work properly on windows may not be that easy.

Once you get `ubuntu.exe` you get everything, including `gpg`. Since I scrapped the assumption that we will not run _any_ Unix script, I'd like to do things exactly as I would do them on any other Unix system.

Reply:
(To be more precise, I am doing things so that working on Windows would be functionally equivalent to working in any other Unix system, except for ~10 lines of code that are Windows specific and configure and make the project using Windows tooling.)

Reply:
Ok, though I think that verifying the package using a hardcoded hash that you have to update, is in any case more robust than importing a key (and adding it to the default keyring) and calling `gpg --verify`. Using this scheme you will consider a package is valid even if it's signed by ANY key that is present inside of the users keyring (and you are also adding a new key that they may not want to their keyring).

Reply:
> verifying the package using a hardcoded hash ... is ... more robust than importing a key

<bikeshed>Technically, I'd say those are **equally** robust as having compromised key in the system keyring brings you compromised CC with `apt-get update && apt-get upgrade`. I don't think that there is major difference between backdoored source code and backdoored compiler.</bikeshed> I'd say both solutions are equally good in terms of _robustness_ and _security_.

Reply:
> Technically, I'd say those are equally robust as having compromised key in the system keyring brings you compromised CC with apt-get update && apt-get upgrade

Well... A part from the fact that the `apt-get` does not use the users keyring, but even if that were the case it would still only be relevant to linux users.

That said, I think the main argument for not using this method is that I am not super happy to have my users keyring contain random keys that I don't care to have in it and some build script modifying the setup of my users home.

If you do decide to keep the pattern of using GPG to verifying (which I don't actually think is a good idea), I would at least provide it some temporary throw-away keyring path with `--keyring --no-default-keyring`.

Reply:
>  I think the main argument for not using this method is that I am not super happy to have my users keyring contain random keys that I don't care to have in it and some build script modifying the setup of my users home.

This is a good concern, and something I did not think about. We could opt for using `signify` &mdash; the native OpenBSD tool for doing these things, which actually works better than GPG as far as solving the problem at hand is the objective. But probably it is simpler if I verify the signtature and then just add the `sha256sum` of such signature in the build system.

I will do that during my next iteration of work on this branch.